cns/service/main.go

```diff
@@ -562,6 +567,18 @@ func main() {
 			return
 		}
 
+		logger.Printf("Starting SyncHostNCVersion")
+		lastUpdatedHostNCTimeStamp := time.Now()
+		rootCxt := context.Background()
+		ctxWithTimeout, _ := context.WithTimeout(rootCxt, cnsconfig.SyncHostNCTimeoutMilliSec)
+		go func() {
+			// Periodically poll vfp programmed NC version from NMAgent
+			for {
+				<-time.NewTicker(cnsconfig.SyncHostNCVersionIntervalMilliSec * time.Second).C
```

Comment:
As discussed instead of making it run periodically, lets make it signal based and trigger as soon as we got an NC update.

Reply:
Need to sync up for it.

Reply:
Will use a new PR to address this feedback.